File: solutions/python/bob/1/bob.py
def response(hey_bob):
    cleaned = hey_bob.strip()
    if not cleaned:
        return "Fine. Be that way!"
    
    is_question = cleaned.endswith("?")
    is_yelling = cleaned.isupper()
    
    if is_yelling and is_question:
        return "Calm down, I know what I'm doing!"
    if is_yelling:
        return "Whoa, chill out!"
    if is_question:
        return "Sure."
    
    return "Whatever."

File: solutions/go/bob/1/bob.go
package bob

import (
    "unicode"
    "strings"
)

func Hey(remark string) string {
    trimmedRemark := strings.TrimSpace(remark)
    if trimmedRemark == "" {
        return "Fine. Be that way!"
    }
    
    hasLetters := false
    allUpper := true
    for _, r := range trimmedRemark {
        if unicode.IsLetter(r) {
            hasLetters = true
            if !unicode.IsUpper(r) {
                allUpper = false
            }
        }
    }
    
    isQuestion := strings.HasSuffix(trimmedRemark, "?")
    switch {
        case hasLetters && allUpper && isQuestion:
        	return "Calm down, I know what I'm doing!"
        case isQuestion:
        	return "Sure."
        case allUpper && hasLetters:
        	return "Whoa, chill out!"
        default:
        	return "Whatever."
    }
}
